src/modules/quoiFeur/quoiFeur.helpers.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  ChannelType,
+  type ChatInputCommandInteraction,
+  Client,
+  Guild,
+  type Message,
+  Role,
+} from 'discord.js';
+
+import { cache } from '../../core/cache';
+import { removeEmoji, removePunctuation } from '../../helpers/regex.helper';
+
+const ONE_MINUTE = 1 * 60 * 1000;
+const MUTED_BY_BOT = 'Muted by bot';
+
+const quoiDetectorRegex = new RegExp(/\b\s*[q][u][o][i]\s*$/i);
```

Comment:
No need to use both the RegExp constructor and a RegExp literal, use just one or the other (generally a literal, the constructor with a string can be useful for clarity when you have a lot of slashes to match but you'll need to be careful with backlash character classes). Character classes that contain just one character are the same as just using that one character. Word boundary marker is redundant with the whitespace check
```suggestion
const quoiDetectorRegex = /\bquoi\s*$/i;
```

From a functional standpoint I don't know that we even want that word boundary, someone ending a sentence with "pourquoi" should get their well deserved "feur".

Reply:
I can totally see something with capture group where if you type `pourquoi` you get back `pourfeur` lmao!!